src/math.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/**
+ * @param array<float> $a
+ * @param array<float> $b
+ */
+function euclidian_dist(array $a, array $b): float
+{
+    assert(count($a) == count($b));
+
+    for ($dist = 0, $n = 0; $n < count($a); $n++) {
+        $dist += pow($a[$n] - $b[$n], 2);
+    }
+
+    return sqrt($dist);
+}
+
+/**
+ * @param array<array<float>> $points
+ * @return array<float>
+ */
+function find_centroid(array $points): array
+{
+    $centroid = [];
+
+    foreach ($points as $point) {
+        foreach ($point as $n => $value) {
+            $centroid[$n] = ($centroid[$n] ?? 0) + $value;
+        }
+    }
+
+    foreach ($centroid as &$value) {
+        $value /= count($points);
+    }
+
+    return $centroid;
+}
+
+/**
+ * @param array<array<float>> $points
+ * @return array{array<float>, array<float>}
+ */
+function find_boundaries(array $points): array
+{
+    $min = [];
+    $max = [];
+
+    foreach ($points as $point) {
+        foreach ($point as $n => $value) {
+            if (!isset($min[$n]) || $min[$n] > $point[$n]) {
```

Comment:
Can be replaced with '$value'
Is there any reason why $point[$n] is better?

Reply:
good point. it's a typo on my side.